NDTensors/test/ITensors/TestITensorDMRG/TestITensorDMRG.jl

```diff
@@ -0,0 +1,14 @@
+## This is getting closer still working on it. No need to review
+## Failing for CUDA mostly with eigen (I believe there is some noise in
+## eigen decomp with CUBLAS to give slightly different answer than BLAS)
+module TestITensorDMRG
+using ITensors
+using NDTensors
+
+reference_energies = Dict([
+  (4, -1.6160254037844384), (8, -3.374932598687889), (10, -4.258035207282885)
+])
+test_tolerance = Dict(([Float32, 1e-5], [Float64, 1e-12]))
```

Comment:
Looks like this can be removed?

Reply:
I wasn't sure if we want to be stricter in the float32 case. the `eps(Float32) * 10^3` is only `0.001`?

Reply:
Hmm yeah good point. I don't think we need to focus on super high precision tests right now since that is tricky anyway, I think we are just trying to catch if things go very wrong (like what this PR is fixing). Let's keep it simple for now (i.e. the current code) and adjust things as we go along.

Reply:
Here is a proposal for a formula for `rtol`:
```julia
julia> default_rtol(elt) = 10^(0.75 * log10(eps(real(elt))))
default_rtol (generic function with 1 method)

julia> default_rtol(Float32)
6.415531155005385e-6

julia> default_rtol(Float64)
1.8189894035458617e-12
```
So basically adjust based on the number of digits of precision.

The goal here is to try to find a general formula so we don't have to hardcode for different number types.

In principle `cutoff` should also be adjusted based on the element type but that's a story for another time, since it is something we would want to look at systematically inside the `svd`/`eigen` functions and elsewhere in the library.

Reply:
Before resolving a conversation could you explain in what way it will be resolved?

Reply:
Sure, I created your `default_rtol` function in the `TestITensorDMRG` file and used it

Reply:
Thanks, it is helpful even if you just say you plan to fix something in your next commit. I use open comments to track what still needs to be done/what I need to check over before merging a PR which becomes difficult if you resolve a conversion without noting down how or if it will be fixed.